jconstraints-core/src/main/java/gov/nasa/jpf/constraints/smtlibUtility/smtsolver/SMTCMDContext.java

```diff
@@ -0,0 +1,213 @@
+    ProcessBuilder pb = new ProcessBuilder(command);
+    pb.redirectErrorStream(true);
+    try {
+      p = pb.start();
+      ctx = new SMTLibExportGenContext(new PrintStream(p.getOutputStream()));
+      bos = new BufferedReader(new InputStreamReader(p.getInputStream()));
+    } catch (IOException e) {
+      e.printStackTrace();
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public void push() {
+    ctx.push();
+    stack.push();
+  }
+
+  @Override
+  public void pop(int n) {
+    ctx.pop(n);
+    stack.pop(n);
+  }
+
+  @Override
+  public Result solve(Valuation result) {
+    try {
+      String output = "";
+      String line;
+      if (!hasCalledAdd) {
+        System.out.println("You might want to add some expressions first");
+      } else {
+        ctx.flush();
+        while (!bos.ready() && p.isAlive()) {
+          Thread.sleep(10);
+        }
+        while (bos.ready() && (line = bos.readLine()) != null) {
+          output += line;
+          output += "\n";
+        }
+      }
+      ctx.solve();
+      String solving = "";
+      while (!bos.ready() && p.isAlive()) {
+        Thread.sleep(10);
+      }
+      while (bos.ready() && (line = bos.readLine()) != null) {
+        output += line;
+        output += "\n";
+      }
+      if (output.contains("Parse Error")) {
+        System.err.println("Something went wrong in the SMT Solver process");
+        System.err.println(output);
+        return Result.ERROR;
+      }
+      Result res = Result.DONT_KNOW;
+      for (String l : output.split("\n")) {
+        if (l.startsWith("<stdin>")) {
+          continue;
+        } else {
+          if (l.equals("sat")) {
+            res = Result.SAT;
+            if (result != null) {
+              List<Variable<?>> lvars = stack.getVarsInStack();
+              SMTCMDSolver.getModel(ctx, bos, lvars, result);
+            }
+          } else if (l.equals("unsat")) {
+            res = Result.UNSAT;
+            if (unsatCoreSolver) {
+              ctx.getUnsatCore();
+              String core = "";
+              while (!bos.ready() && p.isAlive()) {
+                Thread.sleep(10);
+              }
+              while (bos.ready() && (line = bos.readLine()) != null) {
+                core += line;
+                core += "\n";
+              }
+              analyze_core(core);
+            }
+          }
+        }
+      }
+      return res;
+    } catch (IOException | InterruptedException e) {
+      e.printStackTrace();
+    }
+    return Result.ERROR;
+  }
+
+  private void analyze_core(String core) {
+    lastUnsatCore = new LinkedList<>();
+    String[] parts = core.split("\n");
+    for (int i = 1; i < parts.length - 1; i++) {
+      lastUnsatCore.add(namedExpressions.get(parts[i]));
+    }
+  }
+
+  @Override
+  public void add(List<Expression<Boolean>> expressions) {
+    SMTLibExportVisitor visitor = new SMTLibExportVisitor(ctx, smtExportConfig);
+    for (Expression f : expressions) {
```

Comment:
Rawtype (why variable name `f` for an expression?)
```suggestion
    for (Expression<Boolean> f : expressions) {
```